cmd/kubectl.go

```diff
@@ -37,29 +42,73 @@ func runKubectl(ctx *cli.Context) error {
 		return err
 	}
 
-	cluster, err := getClusterByID(c, c.UserConfig.FocusedCluster())
+	config, err := loadConfig(ctx)
 	if err != nil {
 		return err
 	}
 
-	config, err := c.ManagementClient.Cluster.ActionGenerateKubeconfig(cluster)
+	currentRancherServer := config.FocusedServer()
+	if currentRancherServer == nil {
+		return fmt.Errorf("no focused server")
+	}
+
+	currentToken := currentRancherServer.AccessKey
+	t, err := c.ManagementClient.Token.ByID(currentToken)
 	if err != nil {
 		return err
 	}
 
-	tmpfile, err := ioutil.TempFile("", "rancher-")
+	currentUser := t.UserID
+	kubeConfig, err := getKubeConfigForUser(ctx, currentUser)
 	if err != nil {
 		return err
 	}
-	defer os.Remove(tmpfile.Name())
 
-	_, err = tmpfile.Write([]byte(config.Config))
+	var isTokenValid bool
+	if kubeConfig != nil {
+		tokenID, err := extractKubeconfigTokenID(*kubeConfig)
+		if err != nil {
+			return err
+		}
+		isTokenValid, err = validateToken(tokenID, c.ManagementClient.Token)
+		if err != nil {
+			return err
+		}
+	}
+
+	var kubeConfigBytes []byte
```

Comment:
This is only used inside the if block so it probably doesn't need to be created here